hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/sink/utils/Pipelines.java

```diff
@@ -213,12 +215,31 @@ public static DataStream<Object> append(
       Configuration conf,
       RowType rowType,
       DataStream<RowData> dataStream) {
+    Option<Partitioner> insertPartitioner = getInsertPartitioner(conf);
+    DataStream<RowData> finalDataStream;
+
+    if (insertPartitioner.isPresent()) {
+      RowDataKeyGen rowDataKeyGen = RowDataKeyGen.instance(conf, rowType);
+      finalDataStream = dataStream.partitionCustom(insertPartitioner.get(), rowDataKeyGen::getHoodieKey);
+    } else {
+      finalDataStream = dataStream;
+    }
+
     WriteOperatorFactory<RowData> operatorFactory = AppendWriteOperator.getFactory(conf, rowType);
 
-    return dataStream
-        .transform(opName("hoodie_append_write", conf), TypeInformation.of(Object.class), operatorFactory)
-        .uid(opUID("hoodie_stream_write", conf))
-        .setParallelism(conf.getInteger(FlinkOptions.WRITE_TASKS));
+    return finalDataStream
+        .transform(opName("hoodie_append_write", conf), TypeInformation.of(Object.class), operatorFactory);
```

Comment:
```suggestion
        .transform(opName("hoodie_append_write", conf), TypeInformation.of(Object.class), operatorFactory)
        .setParallelism(conf.getInteger(FlinkOptions.WRITE_TASKS));
```

Reply:
Thanks @xuzifu666 , changed. Missed this change Orz..